app/src/main/java/app/revanced/integrations/shared/checks/Check.java

```diff
@@ -0,0 +1,128 @@
+import android.content.Intent;
+import android.net.Uri;
+import android.text.Html;
+import android.text.method.LinkMovementMethod;
+import android.widget.Button;
+import android.widget.TextView;
+import app.revanced.integrations.shared.Logger;
+import app.revanced.integrations.shared.Utils;
+import app.revanced.integrations.youtube.settings.Settings;
+
+import java.util.function.Function;
+
+import static android.text.Html.FROM_HTML_MODE_COMPACT;
+import static app.revanced.integrations.shared.StringRef.str;
+
+abstract class Check {
+    private static final Uri GOOD_SOURCE = Uri.parse("https://revanced.app");
+    private static final Function<Context, AlertDialog.Builder> CHECK_FAILED_DIALOG_BUILDER =
+            (context) -> new AlertDialog.Builder(context)
+                    .setCancelable(false)
+                    .setIcon(android.R.drawable.ic_dialog_alert)
+                    .setTitle(str("revanced_check_environment_failed_title"))
+                    .setPositiveButton(
+                            str("revanced_check_environment_dialog_open_official_source_button"),
+                            (dialog, which) -> {
+                                final var intent = new Intent(Intent.ACTION_VIEW, GOOD_SOURCE);
+                                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+
+                                context.startActivity(intent);
+                            }
+                    ).setNegativeButton(
+                            str("revanced_check_environment_dialog_ignore_button"),
+                            (dialog, which) -> {
+                                final int current = Settings.CHECK_ENVIRONMENT_WARNING_ISSUED_COUNT.get();
+                                Settings.CHECK_ENVIRONMENT_WARNING_ISSUED_COUNT.save(current + 1);
+
+                                dialog.dismiss();
+                            }
+                    );
+
+    private final String failedReasonStringKey;
+
+    /**
+     * @param failedReasonStringKey The string key of the reason why the check failed to be shown to the user.
+     */
+    Check(String failedReasonStringKey) {
+        this.failedReasonStringKey = failedReasonStringKey;
+    }
+
+    protected abstract boolean run();
+
+    static boolean shouldRun() {
+        return Settings.CHECK_ENVIRONMENT_WARNING_ISSUED_COUNT.get() < 2;
+    }
+
+    static void disableForever() {
+        Logger.printDebug(() -> "Environment checks disabled forever.");
+
+        Settings.CHECK_ENVIRONMENT_WARNING_ISSUED_COUNT.save(Integer.MAX_VALUE);
+    }
+
+    @SuppressLint("NewApi")
+    static void issueWarning(Context context, Check... failedChecks) {
+        final var reasons = new StringBuilder();
+
+        reasons.append("<ul>");
+
+        for (var check : failedChecks) {
+            reasons.append("<li>").append(str(check.failedReasonStringKey));
+        }
+
+        reasons.append("</ul>");
+
+        final var finalMessage = Html.fromHtml(
```

Comment:
The links act as a verification method. It's a strategic choice we also use on other places such as the website and social media that certify us as legitimate because we can provide cross links with many followers proving our legitimacy. Ripoff sites can not do that and therefore have no ability to certify themselves with such links. I think it makes sense to keep them there in case someone starts to have it's doubts. Anyone with intention to verify the legitimately of revanced.app could check out the links snd be fully assured that it is indeed the site they probably want to use.